ESP8266/ESP8266.cpp

```diff
@@ -366,15 +374,38 @@ int32_t ESP8266::recv(int id, void *data, uint32_t amount)
     return -1;
 }
 
+nsapi_error_t ESP8266::_oob_socket_closed(const int id)
+{
+    // Packet without payload tells there is no more data coming
+    for (struct packet **p = &_packets; *p; p = &(*p)->next) {
```

Comment:
I don't see why this empty packet is needed. A boolean flag will suffice, as long as you check it after finding the buffer is empty.

      read()
          return next packet if available
          otherwise if queue empty
          if error, return it
          if neatly closed return 0
          otherwise EWOULDBLOCK

read() on a device should always drain its buffers before reporting any connection problems or similar.

Reply:
Alternative solution committed.